### PR TITLE
Add check for menoh api

### PR DESCRIPTION
**Pull request by okdshin**

### CODE REVIEW

File: menoh/menoh_chainer_compiler.cpp [hunk 1]
```diff
@@ -749,6 +759,9 @@ menoh_error_code menoh_model_run(menoh_model_handle model) {
                 auto const& array = chainerx::AsContiguous(output.second->GetArray());
                 auto const& shape = array.shape();
                 auto bytesize = shape.GetTotalSize() * chainerx::GetItemSize(array.dtype());
+                assert(model->variable_profiles.find(output.first) != model->variable_profiles.end());
```
Comment (shinh) [inline menoh/menoh_chainer_compiler.cpp]: CHECK_NE please

File: menoh/menoh_chainer_compiler.cpp [hunk 2]
```diff
@@ -749,6 +759,9 @@ menoh_error_code menoh_model_run(menoh_model_handle model) {
...
+                assert(bytesize == menoh_impl::total_size_in_bytes(variable_profiles.find(output.first)->second) &&
```
Comment (shinh) [inline menoh/menoh_chainer_compiler.cpp]: CHECK_EQ(...) << "allocated output..."

File: menoh/menoh_chainer_compiler.cpp (hunk 1 above)
Comment (okdshin) [reply menoh/menoh_chainer_compiler.cpp]: OK. I will modify.

Comment (okdshin) [reply menoh/menoh_chainer_compiler.cpp]: I can't use CHECK_NE here because iterator is not printable.
I used `CHECK(a != b)` instead of `CHECK_NE(a, b)`

**Pull request merged**